server/conf/evolutions/default/44.sql

```diff
@@ -28,39 +28,46 @@ $$ LANGUAGE SQL;
 
 -- A function that sets the "latest_status" column anytime an application is updated.
 -- Note: The "NEW" variable corresponds to the application record to be inserted / updated.
-CREATE OR REPLACE FUNCTION process_application_change() RETURNS TRIGGER AS $$
+CREATE OR REPLACE FUNCTION process_latest_status_on_application_change() RETURNS TRIGGER AS $$
   BEGIN
-    NEW.latest_status := retrieve_latest_status(NEW.id);;
+    NEW.latest_status := retrieve_latest_application_status(NEW.id);;
     RETURN NEW;;
   END;;
 $$ LANGUAGE plpgsql;
 
-CREATE TRIGGER application_change BEFORE INSERT OR UPDATE ON applications
-    FOR EACH ROW EXECUTE FUNCTION process_application_change();
-
 -- A function that sets the "latest_status" column on the referenced application record anytime
 -- an event record is updated.
 -- Note: The "NEW" variable corresponds to the application event record to be inserted / updated.
-CREATE OR REPLACE FUNCTION process_application_event_change() RETURNS TRIGGER AS $$
+CREATE OR REPLACE FUNCTION process_latest_status_on_application_event_change() RETURNS TRIGGER AS $$
   BEGIN
     IF ((NEW.details->>'event_type') = 'STATUS_CHANGE') THEN
       UPDATE applications
-        SET latest_status = retrieve_latest_status(NEW.application_id)
+        -- Don't always use the status event being modified since we care only about setting the
+        -- latest_status column based on the most recent application event, which isn't necessarily
```

Comment:
Ah, because this happens on updates too we want to find the latest as it might not be the latest?

That's ok, though a little unfortunate as we should never be updating the event log rows, but to cover all the bases I see why its here.

Reply:
Yah, exactly. Unfortunate and probably will never happen, but still necessary.